test-data/unit/check-classes.test

```diff
@@ -339,6 +339,19 @@ class B(A):
     def __new__(cls) -> int:
         return 1
 
+[case testOverride__new__AndCallObject]
+from typing import TypeVar, Generic
+
+class A:
+    def __new__(cls, x: int) -> 'A':
+        return object.__new__(cls)
+
+T = TypeVar('T')
+class B(Generic[T]):
+    def __new__(cls, foo: T) -> 'B[T]':
+        return object.__new__(cls)
```

Comment:
Maybe reveal the type of the return value?

Reply:
Sure, though it is `Any`, since that's what the stubs have for `object.__new__`. (Which is probably fixable in typeshed/fixtures?)